x/observer/keeper/observer_set.go

```diff
@@ -67,17 +72,28 @@ func (k Keeper) RemoveObserverFromSet(ctx sdk.Context, address string) {
 	}
 }
 
+// UpdateObserverAddress updates an observer address in the observer set.It makes sure the updated observer set is valid.
 func (k Keeper) UpdateObserverAddress(ctx sdk.Context, oldObserverAddress, newObserverAddress string) error {
 	observerSet, found := k.GetObserverSet(ctx)
 	if !found {
 		return types.ErrObserverSetNotFound
 	}
+	found = false
 	for i, addr := range observerSet.ObserverList {
```

Comment:
nit: why ObserverList is a slice instead of a map?
With a map we would avoid having duplicates by default and would read/write from it with O(1).

Reply:
Maps are not ordered in go, which might lead to complexities. 
We would need to sort everytime before saving, I think using a list is better

Reply:
Gotcha, I didn't know we need them ordered. Thanks!